avro4s-core/src/main/scala/com/sksamuel/avro4s/SchemaUpdate.scala

```diff
@@ -0,0 +1,9 @@
+package com.sksamuel.avro4s
+
+sealed trait SchemaUpdate
+
+object SchemaUpdate {
```

Comment:
I'd appreciate some docs on this as to why it exists as an ADT rather than a method.
I gather it's being used as a command to update a schema by replacing the namespace or replacing the entire thing (for when we resolve a schema?). What was the rational for not just having some helper method.

Reply:
I will add some docs. Replacing the entire schema happens when the client uses the Encoder / Decoder's `withSchema` method to override the inferred schema with a custom one. The client-provided schema is then decomposed along the magnolia-inferred ADT structure to create updates for field encoders / decoders, using a similar path as for namespace updates. So basically, this ADT exists because namespace updates and full schema updates travel in the same direction (from traits to case classes, and from fields to case classes).